libraries/Microsoft.Bot.Builder.Dialogs/Prompts/CloudOAuthPrompt.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    /// <summary>
+    /// This is replacement for the OauthPrompt that replies on the UserTokenClient it finds in the TurnState.
+    /// </summary>
+    internal class CloudOAuthPrompt : Dialog
+    {
+        private readonly OAuthPromptSettings _settings;
+        private readonly PromptValidator<TokenResponse> _validator;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloudOAuthPrompt"/> class.
+        /// </summary>
+        /// <param name="settings">Additional OAuth settings to use with this instance of the prompt.</param>
+        /// <param name="validator">A custom validator that can be used against Message activities.</param>
+        public CloudOAuthPrompt(OAuthPromptSettings settings, PromptValidator<TokenResponse> validator)
```

Comment:
Since we're null checking below and null validator is fine, is there value in making it optional?

If there is, here is the change so you don't have to code it :)

```suggestion
        public CloudOAuthPrompt(OAuthPromptSettings settings, PromptValidator<TokenResponse> validator = null)
```